File: articles/present-bias-immediate-reward/fortran/present_bias_choice.f90
program present_bias_choice
  implicit none
  real :: beta, delta, delayed_reward, immediate_reward, commitment_cost
  real :: delayed_value, immediate_value
  integer :: delay, choice

  beta = 0.72
  delta = 0.97
  delayed_reward = 300.0
  delay = 12
  immediate_reward = 160.0
  commitment_cost = 70.0

  delayed_value = beta * (delta ** delay) * delayed_reward
  immediate_value = immediate_reward - commitment_cost

  if (delayed_value >= immediate_value) then
    choice = 1
  else
    choice = 0
  end if

  print *, "Synthetic delayed choice under present bias:", choice
end program present_bias_choice

File: articles/present-bias-immediate-reward/rust/present_bias_choice.rs
fn discounted_delayed_value(beta: f64, delta: f64, reward: f64, delay: i32) -> f64 {
    beta * delta.powi(delay) * reward
}

fn choose_delayed_reward(
    beta: f64,
    delta: f64,
    delayed_reward: f64,
    delay: i32,
    immediate_reward: f64,
    commitment_cost: f64,
) -> bool {
    let delayed_value = discounted_delayed_value(beta, delta, delayed_reward, delay);
    let immediate_value = immediate_reward - commitment_cost;
    delayed_value >= immediate_value
}

fn main() {
    println!(
        "Synthetic delayed choice under present bias: {}",
        choose_delayed_reward(0.72, 0.97, 300.0, 12, 160.0, 70.0)
    );
}
